.github/workflows/preview.yml

```diff
@@ -0,0 +1,13 @@
+name: Create preview
```

Comment:
Why you are not using same action library for previews ? I can see they support it https://github.com/jsmrcaga/action-netlify-deploy#use-branch-name-to-deploy even with generating the preview url in the checks section.